src/Drupal/ServiceMap.php

```diff
@@ -53,4 +57,30 @@ public function setDrupalServices(array $drupalServices): void
             }
         }
     }
+
+    private function resolveParentDefinition(string $parentId, array $serviceDefinition, array $drupalServices): array
+    {
+        $parentDefinition = $drupalServices[$parentId] ?? [];
+        if ([] === $parentDefinition) {
+            return $serviceDefinition;
+        }
+
+        if (isset($parentDefinition['parent'])) {
+            if (!isset($drupalServices[$parentDefinition['parent']])) {
+                return $serviceDefinition;
+            }
+
+            return $this->resolveParentDefinition($parentDefinition['parent'], $drupalServices[$parentDefinition['parent']], $drupalServices);
+        }
+
+        if (isset($parentDefinition['class']) && !isset($serviceDefinition['class'])) {
+            $serviceDefinition['class'] = $parentDefinition['class'];
+        }
+
+        if (isset($parentDefinition['public']) && !isset($serviceDefinition['public'])) {
+            $serviceDefinition['public'] = $parentDefinition['public'];
+        }
```

Comment:
I wish we could copy more from the actual dependency injection code. But does it just merge things? Like if we could merge the arrays

For example, this doesn't handle arguments. I guess we don't really inspect those right now.

Base example are plugin managers. Usually a custom class but parent of default plugin manager and the same arguments

Reply:
> I wish we could copy more from the actual dependency injection code. But does it just merge things? Like if we could merge the arrays

I looked at the YamlFileLoader (both on Drupal side and Symfony side) how they did it to make sure I was not making wrong assumptions before coding this.

One of the first thing they do is to either create a `ChildDefinition` if 'parent' is set or `Definition` if it is not.

Then it will define the different properties (class, shared, synthetic, lazy, public, abstract, arguments etc.. etc..) like

```php
if (isset($service['class'])) {
  $definition->setClass($service['class']);
}
```

As said in #208, I am experiencing  with dumping the container to see how it could help with everything (#247, #249 etc..)

> For example, this doesn't handle arguments. I guess we don't really inspect those right now.
>
> Base example are plugin managers. Usually a custom class but parent of default plugin manager and the same arguments

Yep, since arguments are not supported with the actual `DrupalServiceDefinition` and that we don't use them for the moment, I thought about doing another PR later to support arguments.